pkg/apis/serving/v1alpha2/kfservice_validation.go

```diff
@@ -80,10 +81,27 @@ func validateKFService(kfsvc *KFService) error {
 	return nil
 }
 
-func validateModelSpec(spec *PredictorSpec) error {
-	if spec == nil {
+func validateEndpoint(endpoint *EndpointSpec) error {
+	if endpoint == nil {
 		return nil
 	}
+	if err := validatePredictor(endpoint.Predictor); err != nil {
+		return err
+	}
+	if endpoint.Transformer != nil {
+		if err := endpoint.Transformer.Validate(); err != nil {
+			return err
+		}
+	}
+	if endpoint.Explainer != nil {
+		if err := endpoint.Explainer.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validatePredictor(spec PredictorSpec) error {
```

Comment:
this can be moved to a separate file `predictor.go` to be consistent with transformer and explainer?
```
func (t * PredictorSpec) Validate() error
```

Reply:
I was going to punt on this. But I'll do it now.